### PR TITLE
Replace anonymous class creation with lambdas in ConstructorConstructor

### PR DESCRIPTION
**Pull request by panic08**
I think we should change the complete creation of anonymous classes to lambdas, which will reduce the amount of code and make it more readable

Since we recently got Java 8 support, we can afford to use lambdas

### CODE REVIEW

**Comment by panic08** [comment]
Hmm, I noticed odd that the `Build / GraalVM Native Image test (pull_request)` job crashes with text:
```
Run graalvm/setup-graalvm@22cc13fe88ef133134b3798e128fb208df55e1f5
Error: Failed to download graalvm-jdk-17_linux-x64_bin.tar.gz. Are you sure java-version: '17' is correct?
```
I assume it's not related to the change?

**Comment by Marcono1234** [comment]
> Hmm, I noticed odd that the `Build / GraalVM Native Image test (pull_request)` job crashes with text

Seems to be a bug with that GitHub action; I have created #2764 to update it a fixed version. Once that is merged you can merge `main` into your branch and the build should succeed again.

**Review by Marcono1234 — COMMENTED** [review]
Thanks for the changes! I have one small suggestion, see review comment.

Though @eamonnmcmanus might have additional suggestions or requests for changes.

File: gson/src/test/java/com/google/gson/internal/ConstructorConstructorTest.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ private interface Interface {}
   public void testGet_AbstractClassNoArgConstructor() {
     ObjectConstructor<AbstractClass> constructor =
         constructorConstructor.get(TypeToken.get(AbstractClass.class));
-    var e = assertThrows(RuntimeException.class, () -> constructor.construct());
+    var e = assertThrows(RuntimeException.class, constructor::construct);
```
Comment (eamonnmcmanus) [inline gson/src/test/java/com/google/gson/internal/ConstructorConstructorTest.java:45]: Unless Error Prone is making you do this, I think this code should probably be left the way it is. It's asserting that the operation `constructor.construct()` throws an exception, and I think `() -> constructor.construct()` is a clearer way to express that.

**Review by eamonnmcmanus — APPROVED** [review]

**Pull request merged**